### DISCUSSION TITLE
Roast of Sentry

### DISCUSSION THREAD

**Discussion by rahulchhabria** [getsentry/sentry#79336] [Problem Understanding]
It’s roast time. In honor of our ego being too big, we’re asking you to knock us back down to size with a little feedback. Tell us everything you hate (or find mildly annoying) about our product. After that crushing blow, we’ll try our best to fix it. 

To leave feedback, create a new issue in The Roast of Sentry discussion.

More details about the roast coming soon.

**Comment by jalaziz** [comment]
Well for one, the link 404s 😄

**Reply by rahulchhabria** [reply]
🤣 you got me. adding this discussion was on my to-do. was going to publish the actual roast site tomorrow. timing is hard 😔. but stay tuned. more coming soon.

**Reply by shaedrich** [reply]
Admit it, it was bait 😜

**Reply by rahulchhabria** [reply]
lol i wish i was that crafty. instead, i was like... oh no i better do this before i forget. 😆 but it's up now. check it out. we'll post more on our social channels next week.

**Reply by shaedrich** [reply]
We've all been there. Good to know that you folks over there at Sentry, who do great work, are only human after all 😉

**Comment by jalaziz** [comment]
Getting ahead of the site being up, the thing we find most annoying about the product are already issues / discussions:
* https://github.com/getsentry/sentry/issues/27725
* https://github.com/getsentry/sentry/issues/54732

Basically, the amount of alert rule duplication needed when using multiple services and environments is a massive PITA without automation.

**Comment by max-wittig** [comment]
I really appreciate Sentry for what it is and we offer it to internal employees (only internally!) with a multi-org setup
and using the self-hosted version, so the following:

* No OIDC login, with a provider that we specify in the admin setup (no OpenID!!!) so we created https://github.com/siemens/sentry-auth-oidc, which is used by many people. Sadly this causes the people in the whole instance to be able to login as member in all orgs so we had to limit the rights of `members` in organizations to zero.
* Custom patches in-order to automate Sentry for user management so that we can on-board people based on the groups they use in Gitlab
* Custom patches to disable functionality that is not working inside our environment (air-gapped) like slack, but that people still like to configure and try, which messes Sentry up
* Mostly a pain to update, no readable changelogs, just dumps of commits in the release text without knowing the actual changes
* No way to figure out, which organization is causing Sentry to use 100% CPU. Lack of monitoring
* A release way yanked, which had severe consequences and caused up to drop random data in the DB until we could restore our backup again: https://github.com/getsentry/self-hosted/issues/3167
* We had to block certain nginx routes, so that other employees are not able to e.g. disconnect our OIDC provider (and complain that they can't login), increase rights of owners
* 55 containers running
* We would love to offer Sentry to more internal employees, but its just not stable and reliable enough for that regarding updates and monitoring

**Reply by frederikspang** [reply]
Goes against the 55 containers running - But default self-hosted default setup gave me connection problems with postgresql; Until I added pgbouncer to the mix.

**Comment by hwittenborn** [comment]
Also until the link manages to come back online...

It would be nicer if it was easier to deploy Sentry. I understand the need for multiple containers since there's so much Sentry needs to do what it does. But it would be *really* nice if I could simply have a Docker Compose file that didn't need updated via a script. Not sure what this would look like in terms of needed changes, but I would definitely prefer it more 😄.

**Reply by naanlizard** [reply]
The docker compose file for self hosted sentry is an affront against man and god alike

**Comment by stephenlacy** [comment]
Native sourcemaps! Very tedious to navigate sourcemaps and dumps for Unreal Engine.

**Reply by bruno-garcia** [reply]
Dealing with native debug files is definitely tedious. That said we're always on a look out in ways we can make it less so.
Could you share some of your pain points? How do you manage it today?

What platforms are you building for? Windows, Android?

About 'navigating dumps' for Unreal could you expand what you mean by that?

**Reply by stephenlacy** [reply]
The debug symbols were not matched correctly, they seemed to not point to the error lines.

When working in UE we would love to use it for our errors such that things are divided by:
- Editor: when people are in the editor and they encounter warnings/errors/asserts (Unreal Ensures) that they are tracked
- Game: during playtests, warnings/errors/ensures are tracked

Viewing these errors with callstacks would be a great boost to productivity over the current approach which is:
- take screenshot of the warning/error/ensure
- copy/paste the text (which not a lot of people do)
- describe what you were doing

With the above, we still don't know the frequency of it during Editor sessions and Playtests.

**Reply by bruno-garcia** [reply]
> The debug symbols were not matched correctly, they seemed to not point to the error lines.

Could you please open a ticket about this? A repro would be helpful. I wonder if it's a UE build tool issue or what could be.

> Editor: when people are in the editor and they encounter warnings/errors/asserts (Unreal Ensures) that they are tracked
> Game: during playtests, warnings/errors/ensures are tracked

We add a tag OOTB for editor vs game. Is it hard to tell apart crashes from one vs the other?

> Viewing these errors with callstacks would be a great boost to productivity over the current approach which is:

Sorry I don't follow, which errors don't have callstacks? 

Are you using the Unreal Engine SDK? https://github.com/getsentry/sentry-unreal
If so, which version of the SDK?
Or the [Crash Report Client](https://docs.sentry.io/platforms/unreal/configuration/setup-crashreporter/)?

Also, which platforms are you targeting?
I ask because different 'backends' run depending on the platform.

Which UE Version are you on? 

Please feel free to open tickets on https://github.com/getsentry/sentry-unreal so we can debug any issues you're having.
You shouldn't have to look at screenshot to debug issues and if line numbers don't point to the right place, we should be looking at that too.

Thank you for your roast ! :) we can always improve things, with your help

**Comment by catdogmat** [comment]
Metrics!

They looked like exactly what we needed, but then the beta ended.

**Reply by md384** [reply]
Same! We were looking for a way to get metrics from our web clients and sentry metrics were working great

**Reply by jvineet** [reply]
The custom metrics feature was a pretty great "push-metrics" solution, especially considering the ease of sending metrics from internal svcs as well as browsers/web clients etc.

**Reply by manishrawat1992** [reply]
Integration of metrics with traces doesnt seem like a good idea. Traces is not a feature everybody requires especially for react native

The Metrics product was very good. Currently we are sending errors to compute count metric.

**Comment by paulmiu** [comment]
Self-hosting Sentry is a pain!
Give the on-prem community more love.
There are still tons of ways to monetize on-prem solutions. (Look at Odoo for instance)
I believe it'll grow your userbase significantly, because devs love Sentry!

**Comment by frederikspang** [comment]
More utilities for CPU and Memory usage! Allocations and CPU Time stats per span, to help diagnose performance even more! (Just a convention for it; https://develop.sentry.dev/sdk/telemetry/traces/span-data-conventions/)

**Comment by jclark-uaudio** [comment]
Support!  
Zendesk is still in read-only mode and emails and tickets get no response.  Is there a new method I'm unaware of?

**Reply by souredoutlook** [reply]
hey @jclark-uaudio 👋 Senior Support Manager here! 

Would love to hear more about this - I looked and I do see a recent ticket (April) that can be attributed to you.

Would it be okay if I reach out?

P.S. Big fan of your software (and hardware too!)

**Reply by jclark-uaudio** [reply]
Hey @souredoutlook ,

Glad to hear from a fan!  I was actually able to figure out the SCIM integration with Sentry.  I forget exactly how I solve it but all the documentation online was outdated.

Thanks!

**Comment by MattisAbrahamsson** [comment]
Sentry is great. 
I guess if I have to roast something... I think the issue filtering needs improvement. Being able to write dynamic filters such as any issues in the latest version etc. Also being able to share the custom views with the whole team is something I am missing a lot.
Thanks!

**Reply by rahulchhabria** [reply]
Thanks for the feedback and for participating in the Roast. I owe you a prize. Can you email me at rahul at sentry dot io, please?

**Comment by pinkeyAT** [comment]
What i have issues with is that it ask way to often for my second factor. 
I use login with google and no other site ask that often for the 2FA, its really annoying. 
Testet it with chrome and safari and for both browsers the same issue.

**Reply by rahulchhabria** [reply]
Thanks for the feedback and for participating in the Roast. I owe you a prize. Can you email me at rahul at sentry dot io, please?

**Reply by pinkeyAT** [reply]
> Thanks for the feedback and for participating in the Roast. I owe you a prize. Can you email me at rahul at sentry dot io, please?

Hi, did you receive my mail? Haven’t got a response. Thanks :) @rahulchhabria

**Reply by rahulchhabria** [reply]
oh man! sorry i missed it. just replied!

**Comment by koen-dev** [comment]
It was hard to find something to roast because Sentry is generally great, but hey—a roast is a roast.

How come the UI is so inconsistent when it comes to wrapping filters and toolbar items on mobile? For example, in Insights > Frontend > Web Vitals, the Browsers filter refuses to wrap to the next line, while other filters have no problem doing so. Are these elements in some sort of flexbox civil war?

And what’s the deal with the question mark tooltips getting completely overlapped by the card titles (LCP, FCP, etc.)? If I have to play hide-and-seek with a tooltip, I’m just going to assume the answer is “¯\(ツ)/¯”.

Oh, and let’s talk about that Poor score red text on a dark red background. I appreciate the dedication to the “red means bad” theme, but it doesn’t exactly scream accessible contrast. More like “hope you brought night vision goggles” levels of readability.

That said, you’re still worthy enough for me to rock multiple Sentry stickers on my laptop—because nothing says “I embrace the chaos” like debugging frontend issues in production.

**Comment by torickjdavis** [comment]
Coming here from the [X post](https://x.com/getsentry/status/1889438859957809153). Replays are honestly one of my favorite features, ever. But when you're on the go and check one out on mobile, the replay experience is not great; landscape or portrait. Having a way to adjust the view of the user's screen or even to just make it playback fullscreen would be incredible. The tiny draggable bars just don't quite do the trick.

**Comment by bruno-garcia** [comment]
No Session Replay on the Blazor SDK 👻

**Comment by MattisAbrahamsson** [comment]
For my UI shaming, I just stumbled on this wonderful error toast today while trying to enable an uptime monitoring alert
![Screenshot 2025-02-13 at 19 36 42](https://github.com/user-attachments/assets/0db2aabb-2c80-4de2-b8af-74a38b7a9ce5)
Y indeed...

**Reply by souredoutlook** [reply]
Hey Mattis 👋  - Support Manager for NA here. 

 I've noted some additional frustrations around this from other users - I've escalated to @gaprl and team

Are you still hitting this problem? Let me know as I believe the underlying cause is due to no available Pay-as-you-go budget

**Reply by MattisAbrahamsson** [reply]
Hey, yeah I figured out the reason, I could see the entire error when I opened devtools and looked at the response 😄 It worked fine after increasing the budget. But the error message could be better 😅

**Reply by souredoutlook** [reply]
Hi @MattisAbrahamsson the fix for this message is being applied here: https://github.com/getsentry/sentry/pull/85574

Appreciate you raising this

**Comment by md384** [comment]
We were very excited to try profiling but it turned out to not be very useful for us because the sampling rate was too low. I wrote a comment in https://github.com/getsentry/sentry-javascript/discussions/12446#discussioncomment-9756243 about how it didn't work for us.

I would love to be able to set the sampling rate (something that is available in the V8 profiler). I understand that might be problematic for hosted sentry since it will increase storage needs but we use sentry self-hosted so can tune our storage usage to support high resolution profiles.
